src/block_pantry/block_ckpt.c

```diff
@@ -15,20 +15,24 @@
  *     checkpoint again).
  */
 static int
-__bmp_checkpoint_pack_raw(
-  WT_BLOCK_PANTRY *block_pantry, WT_SESSION_IMPL *session, WT_ITEM *root_image, WT_CKPT *ckpt)
+__bmp_checkpoint_pack_raw(WT_BLOCK_PANTRY *block_pantry, WT_SESSION_IMPL *session,
+  WT_ITEM *root_image, WT_PAGE_BLOCK_META *block_meta, WT_CKPT *ckpt)
 {
     uint64_t pantry_id;
     uint32_t checksum, size;
     uint8_t *endp;
 
+    WT_ASSERT(session, block_meta != NULL);
+    WT_ASSERT(session, block_meta->page_id != WT_BLOCK_INVALID_PAGE_ID);
+
     /*
      * !!!
      * Our caller wants the final checkpoint size. Setting the size here violates layering,
      * but the alternative is a call for the btree layer to crack the checkpoint cookie into
      * its components, and that's a fair amount of work.
      */
-    ckpt->size = __wt_atomic_loadv64(&block_pantry->next_pantry_id);
+    /* ckpt->size = __wt_atomic_loadv64(&block_pantry->next_pantry_id); */
+    ckpt->size = block_meta->page_id; /* XXX What should be the checkpoint size? Do we need it? */
```

Comment:
Good question - can we use the root page size? Since we don't have the extent list business for this block manager.

Reply:
Yes, a good question.  We should figure out how it's used, that should inform what to put in it.  Is it only informational for statistics?